src/main/java/bio/terra/service/dataset/DatasetTableDao.java

```diff
@@ -138,17 +120,12 @@ private void createColumns(UUID tableId, Collection<Column> columns) {
                 .collect(Collectors.toList());
             table.primaryKey(naturalKeyColumns);
 
-            String modeName = rs.getString("partition_mode");
-            if (modeName.equals(DatePartitionMode.NAME)) {
-                table.partitionMode(new DatePartitionMode(rs.getString("partition_column")));
-            } else if (modeName.equals(IntPartitionMode.NAME)) {
-                String column = rs.getString("partition_column");
-                long min = rs.getLong("int_partition_min_value");
-                long max = rs.getLong("int_partition_max_value");
-                long interval = rs.getLong("int_partition_interval");
-                table.partitionMode(new IntPartitionMode(column, min, max, interval));
-            } else {
-                table.partitionMode(NoPartitionMode.INSTANCE);
+            String bqPartitionConfig = rs.getString("bigquery_partition_config");
```

Comment:
We need to understand how we version this. If we need to add/change the json schema, how do we do it?

One idea would be:
1. Store this in a jsonb column (you have it as json), so it is queryable
2. Add a version string as a member of the BigQueryPartitionConfig class
3. To deserialize, read the version out of the json and read the json as a string
4. Based on the version, use objectMapper to map to the matching class.




serialize by retrieving the version